File: POO/Slide03/Conta.java

public class Conta {
	public String numero;
	public String cpf;
	public double saldo;
	
	public Conta(String numero, String cpf) {
		this.numero = numero;
		this.cpf = cpf;
		this.saldo = 0;
	}
	
	public void creditar(double valor) {
		saldo += valor;
	}
	
	public void debitar(double valor) {
		saldo -= valor;
	}
	
	public void transferir(double valor, Conta conta) {
		this.debitar(valor);
		conta.creditar(valor);
	}
	
	public boolean vazia() {
		if (saldo == 0) {
			return true;
		}else {
			return false;
		}
	}
	
	public Conta clonar() {
		Conta c2 = new Conta(numero, cpf);
		c2.creditar(saldo);
		return c2;
	}

	@Override
	public String toString() {
		return "Conta [numero=" + numero + ", cpf=" + cpf + ", saldo=" + saldo + "]";
	}
	
	
}

File: PWEB1/aprendizado typescript/Projeto bancário em Typescript/banco-node-1/app/ts/model/conta.ts
class Conta {
    private _numero: string;
    private _saldo: number;
    constructor( numero:string, saldo:number=0) {
        this._numero = numero;
        this._saldo = saldo;
    }

    get numero() {
        return this._numero;
    }

    set numero(numero) {
        this._numero = numero;
    }

    get saldo() {
        return this._saldo;
    }

    debitar(valor:number) {
        //apenas debita se houver saldo
        if (valor < this._saldo) {
            this._saldo -= valor;
        }
    }

    creditar(valor:number) {
        this._saldo += valor;
    }

}
